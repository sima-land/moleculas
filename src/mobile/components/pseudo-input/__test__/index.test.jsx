import React from 'react';
import { shallow } from 'enzyme';
import { act } from 'react-dom/test-utils';
import PseudoInput from '../';

describe('<PseudoInput />', () => {
  it('should render without props', () => {
    const component = shallow(<PseudoInput />);

    expect(component).toMatchSnapshot();
  });

  it('should handle onClick prop', () => {
    const onClick = jest.fn();
    const component = shallow(<PseudoInput onClick={onClick} />);

    expect(onClick).toBeCalledTimes(0);
    act(() => {
      component.find('.input').prop('onClick')();
    });
    expect(onClick).toBeCalledTimes(1);
  });
});
