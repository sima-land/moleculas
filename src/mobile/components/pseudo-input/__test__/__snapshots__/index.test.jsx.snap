// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PseudoInput /> should render without props 1`] = `
<div
  className="input"
  role="button"
>
  <span
    className="icon"
  >
    <ForwardRef(MagnifierSVG)
      aria-hidden="true"
      fill="#9e9e9e"
    />
  </span>
  <span
    className="text"
  />
</div>
`;
